### PR TITLE
Add Sphinx support for generating man pages

### LINKED ISSUE #103: Documentation doesn't support generating man pages
Up to now I've been generating man page as documentation using setuptools `build_sphinx` command.
Looks like in l;atest version sphinx support has been removed (no dosc/copy.py files).
How can I now generate documentation as man page?

```console
[tkloczko@ss-desktop flexmock-0.11.1]$ /usr/bin/python3 setup.py build_sphinx -b man --build-dir build/sphinx
running build_sphinx
Running Sphinx v4.2.0

Configuration error:
config directory doesn't contain a conf.py file (/home/tkloczko/rpmbuild/BUILD/flexmock-0.11.1)
```

### PR DESCRIPTION
**Pull request by ollipa**
Re-added Sphinx support so that man pages can be generated from the documentation. `myst-parser` needs to be installed so that Sphinx can understand markdown files:

```
pip install --upgrade myst-parser
```

HTML documentation is still generated using Mkdocs.

Closes #103

### CODE REVIEW

File: docs/toctree.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+   usage/argument_matching
+   advanced/builtin_functions
+   advanced/call_order
+   advanced/chained_methods
+   advanced/generators
+   advanced/multiple_return
+   advanced/new_instances
+   advanced/replacing_functions
+   advanced/state
+   compare
```
Comment (christophe-riolo) [inline docs/toctree.rst:21]: Is there no way to generate this ? I'm afraid it's quickly going to be outdated :)

**Review by christophe-riolo — APPROVED** [review]
Good as-is, just one comment, and also, one question can't we add the myst parser to the dev dependencies in poetry rather ?

**Comment by ollipa** [comment]
> Good as-is, just one comment, and also, one question can't we add the myst parser to the dev dependencies in poetry rather ?

Yeah I guess it wouldn't hurt to add Sphinx and myst parser to the dependencies 👌

File: docs/toctree.rst (hunk 1 above)
Comment (ollipa) [reply docs/toctree.rst:21]: Yeah, I don't like having duplicate table of contents here either. I'll look into it. Maybe we could test in CI that Sphinx build doesn't raise any warnings or errors.

Comment (ollipa) [reply docs/toctree.rst:21]: I added CI job to test Sphinx documentation generation:
https://github.com/flexmock/flexmock/pull/105/commits/8b39f661ac0b1bed36aca1cfc098965d72466ccb

**Pull request merged**